paella-matterhorn/javascript/modified-classes/videoloader-for-paella41.js

```diff
@@ -181,8 +181,61 @@ var MHVideoLoader = Class.create(paella.VideoLoader, {
     if (presenter) { this.streams.push(presenter); }
     if (presentation) { this.streams.push(presentation); }
 
-    // Callback
-    this.loadStatus = true;
-    onSuccess();      
+    // Get captions.
+    var captionURL = getCaptionURL();
+    if (captionURL) {
+      loadCaptions(captionURL, concludeLoading);
+    }
+    else {
+      setTimeout(concludeLoading, 0);
+    }
+
+    // Like MHCaptionsDataDelegate#read, except it stops when it finds the url
+    // and leaves loading the url up to other functions (like
+    // paella.captions.Caption#initialize and reload).
+    function getCaptionURL() {
+      var catalogs = paella.matterhorn.episode.mediapackage.metadata.catalog;
+      if (!(catalogs instanceof Array)) {
+        catalogs = [catalogs];
```

Comment:
Heh, I don't have a great reason, but I have a reason: This was lifted from `MHCaptionsDataDelegate`, which I wanted to use whole, but doesn't work for this situation because it does extra stuff beyond just extracting the url. If we had tests, I'd start making this less weird. However, I don't know exactly why it defends itself against this situation, so I can't write a meaningful test.